connections/sim/hw/hw_sim.py

```diff
@@ -32,23 +42,45 @@ def __init__(
 
         self._paused = False
 
+    def set_pin_mode(self, pin, mode):
+        """
+        :param pin: Should be a test pin
+        :param mode: The mode you want to set the pin to -> 0 for INPUT and 1 for OUTPUT
+        """
+        with self._lock:
+            self._pin_modes[pin] = mode
+            LOGGER.debug(f"Pin mode of pin={pin} set to={mode}")
+
+    def get_pin_mode(self, pin):
+        """
+        :param pin: Should be a test pin
+        """
+        with self._lock:
+            val = self._pin_modes[pin]
+            LOGGER.debug(f"Pin mode read from pin={pin} returned value={val}")
+            return val
+
     def digital_write(self, pin, val):
         """
         :param pin: Should be a test pin.
         :param val: True to set high, False to set low
         """
         with self._lock:
+            assert self._pin_modes[pin] == PinModes.INPUT
             LOGGER.debug(f"Digital write to pin={pin} with value value={val}")
             if pin in self._ignitor_tests:
                 self._ignitor_tests[pin].write(val)
             elif pin in self._ignitor_fires and val:
                 self._ignitor_fires[pin].fire()
 
+
+
     def analog_read(self, pin):
         """
         :param pin: Should be a read pin. Don't rely on behaviour if the pin isn't a readable pin.
         """
         with self._lock:
+            assert self._pin_modes[pin] == PinModes.OUTPUT
```

Comment:
no need to assert pin mode on analog read

Reply:
Fixed this, but pin mode is still being set to OUTPUT on _handleAnalogRead.  Should I also change this?  It doesn't have any effect other than adding to the pin_modes dictionary in hw_sim

34a53f5